tests/spec/fs.rename.spec.js

```diff
@@ -151,4 +151,44 @@ describe('fs.rename', function() {
       });
     });
   });
+
+  it('(promise version) should rename an existing file', function(done) {
+    var complete1 = false;
+    var complete2 = false;
+    var fs = util.fs();
+
+    function maybeDone() {
+      if(complete1 && complete2) {
+        done();
+      }
+    }
+
+    fs.open('/myfile', 'w+', function(error, fd) {
+      if(error) throw error;
+
+      fs.close(fd, function(error) {
+        if(error) throw error;
+
+        fs.promises.rename('/myfile', '/myotherfile').then(
+          function(){
```

Comment:
With promises, we should be able to get rid of this hack, and instead use [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

Reply:
Implemented Promise.all() version in 5abefb6c6a467eb0271caddc0685760427fd57bc. Using promises certainly cleaned up the code.